src/components/AddAsset/AddAssetDetailsCard.vue

```diff
@@ -84,6 +92,15 @@ export default class AddAssetDetails extends Mixins(TranslationMixin) {
 
   async handleAddAsset(): Promise<void> {
     this.$emit('add');
+    const asset: Partial<Asset> = this.asset || {};
```

Comment:
I think it's better to transfer adding asset logic to mixin from this component. Since component is reused in polkaswap, in one case asset shouldn't be added to account assets (this is the case, then user is not logged in).

```
async handleAddAsset(): void {
  this.$emit('add');
}
```

In parent views you still can listen this event and adding asset